app/containers/popup.tsx

```diff
@@ -30,73 +26,59 @@ const isChromeUrl = (url: string) => {
   return CHROME_URL_REGEX.test(url);
 };
 export class Popup extends React.Component<POPUP_PROPS & DispatchProps, {}> {
-  componentWillMount() {
-    this.props.updateField("interceptStatus", "");
-    this.props.updateField("tabId", this.props.tabId);
-    this.props.updateField("tabUrl", this.props.tabUrl);
-    MessageService.getRequests(this.props.tabId, requests => {
-      this.props.updateField("requests", requests);
-    });
-
-    MessageService.getEnabledStatus(this.props.tabId, (enabledStatus: boolean) => {
-      this.props.updateField("enabled", enabledStatus);
-    });
-    this.props.updateField("errorMessage", "");
-    if (this.props.isInterceptorOn[this.props.tabId] === undefined) {
-      this.props.updateInterceptorStatus(this.props.tabId, true);
-    }
-  }
-
   isUrlInValid = (tabUrl: string) => {
     return !tabUrl || isChromeUrl(tabUrl);
   };
 
   handleClick = (_: React.MouseEvent<HTMLButtonElement>): void => {
-    if (this.isUrlInValid(this.props.tabUrl)) {
-      this.props.errorNotify(`Cannot Start Listening on ${this.props.tabUrl}`);
+    const props = this.props;
+    if (this.isUrlInValid(this.props.currentUrl)) {
+      props.errorNotify(`Cannot Start Listening on ${this.props.currentUrl}`, props.currentTab);
       return;
     }
-    if (this.props.enabled) {
-      MessageService.disableLogging(this.props.tabUrl, this.props.tabId);
-      this.props.updateField("enabled", false);
+    if (props.data.enabledStatus) {
+      MessageService.disableLogging(props.currentTab);
     } else {
-      MessageService.getRequests(this.props.tabId, requests => {
-        MessageService.enableLogging(this.props.tabUrl, this.props.tabId);
-        this.props.updateFields({ enabled: true, requests });
-      });
+      MessageService.enableLogging(props.currentTab);
     }
   };
 
   clearRequests = (_: React.MouseEvent<HTMLButtonElement>): void => {
-    MessageService.clearData(this.props.tabId);
-    this.props.clearFields();
+    this.props
+      .clearFields(this.props.currentTab)
+      .then(() => {
+        MessageService.setBadgeTextToZero(this.props.currentTab);
+      })
+      .catch((err: any) => {
+        console.log(err);
+      });
   };
 
-  handleCheckToggle = (reqId: number, presentCheckedState: boolean): void => {
-    this.props.handleCheckToggle(reqId, presentCheckedState);
+  handleCheckToggle = (tabId: number, reqId: number, presentCheckedState: boolean): void => {
+    this.props.handleCheckToggle(this.props.currentTab, reqId, presentCheckedState);
   };
 
-  handleRespTextChange = (value: string, requestId: string): void => {
-    this.props.handleRespTextChange(value, requestId);
+  handleRespTextChange = (value: string, requestId: string, tabId: number): void => {
```

Comment:
I feel this indirection can be avoided. Any reason not to call the `this.props.handle...` function wherever it's used?